### PR TITLE
feat(testing): add harness tests for tag component

### PR DESCRIPTION
**Pull request by zarghamkhandev**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

Partially implements https://github.com/Tinkoff/taiga-ui/issues/476

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/testing/core/tag.harness.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    async getCrossIcon(): Promise<TuiSvgHarness | null> {
+        const svg = await this.locatorForOptional(
+            TuiSvgHarness.with({selector: `.t-icon`}),
+        )();
+
+        return svg;
+    }
+
+    async getTagDiv(): Promise<TestElement> {
```
Comment (waterplea) [inline projects/testing/core/tag.harness.ts:17]: This sounds a bit too low level/implementation detail, I wouldn't want to expose that. If it is only used for color test maybe we can test it some other way?

Comment (zarghamkhandev) [reply projects/testing/core/tag.harness.ts:17]: providing just background color might be an option for now ?

Comment (zarghamkhandev) [reply projects/testing/core/tag.harness.ts:17]: @waterplea can i have your opinion here.

Comment (waterplea) [reply projects/testing/core/tag.harness.ts:17]: Yes, let's go with background color only. We are not publishing this package yet so we might have breaking changes in future if we need to.

Comment (zarghamkhandev) [reply projects/testing/core/tag.harness.ts]: done ✅

**Review by splincode — APPROVED** [review]

**Comment by splincode** [comment]
@zarghamkhandev please, rebase with main branch

**Comment by zarghamkhandev** [comment]
@splincode  done ✅

**Comment by splincode** [comment]
image

**Comment by splincode** [comment]
image

**Review by waterplea — APPROVED** [review]

**Pull request merged**